game.exs

```diff
@@ -1,7 +1,52 @@
 defmodule GuessingGame do
+  @moduledoc """
+  A simple Elixir-based guessing game for guessing a number between 1 and 100.
+  """
 
-  def hi() do
-    IO.puts("hello world")
+  @ver 0.1
+
+  def runner do
+    start_game
+    answer = generate_answer
+    game_loop(answer)
+  end
+
+  defp game_loop(answer) do
+    get_user_guess
+    |>check_user_answer(answer)
   end
 
+  defp start_game do
+    IO.puts("-=Fun Elixir Guessing Game=-")
+    IO.puts("I'll pick a number between 1 and 100 and you try to guess it.")
+  end
+
+  defp generate_answer do
+    :rand.uniform(100)
+  end
+
+  defp get_user_guess do
+    guess = IO.gets "What is your guess?\n"
```

Comment:
I think you could forego assigning the `guess` variable and just pipe the result of IO.gets into String.replace